bolt/src/Passes/ReorderUncondJump.h

```diff
@@ -0,0 +1,57 @@
+//===--------- Passes/ReorderUncondJump.h ---------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_TOOLS_LLVM_BOLT_PASSES_REORDERUNCONDJUMP_H
+#define LLVM_TOOLS_LLVM_BOLT_PASSES_REORDERUNCONDJUMP_H
+
+#include "BinaryPasses.h"
+
+// This pass founds casses when BBs have layout:
+// #BB1:
+// cmp
+// cond_jmp out
+// #BB2:
+// <code>
+// jmp #BB1
+// #BB3 <out>
+//
+// And swaps BB1 and BB2:
+// #BB1:
+// jmp #BB3
+// #BB2:
+// <code>
+// #BB3:
+// cmp
+// cond_jmp #BB2
+// #BB4 <out>
+//
+// And vice versa depending on the profile information.
+// The advantage is that the loop uses only one conditional jump,
+// the unconditional jump is only used once on the loop start.
+
+namespace llvm {
+namespace bolt {
+
+class ReorderUncondJump : public BinaryFunctionPass {
+public:
+  explicit ReorderUncondJump() : BinaryFunctionPass(false) {}
+
+  const char *getName() const override { return "ReorderUncondJump"; }
```

Comment:
Here, instead of "ReorderUncondJump", we should use the flag name used to control this pass execution, thus, "uncond-jmp-reorder".